packages/core/contracts/cross-chain-oracle/chain-adapters/Nomad_ChildMessenger.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../external/nomad/interfaces/XAppConnectionManagerInterface.sol";
+import "../interfaces/ChildMessengerInterface.sol";
+import "../interfaces/ChildMessengerConsumerInterface.sol";
+import "../../common/implementation/Lockable.sol";
+import "../../oracle/implementation/Constants.sol";
+import "../../common/implementation/HasFinder.sol";
+
+/**
+ * @notice Sends cross chain messages from any network where Nomad bridging infrastructure is deployed to L1. Both L1
+ * and the network where this contract is deployed need to have Nomad Home + Replica contracts to send and receive
+ * cross-chain messages respectively.
+ */
+contract Nomad_ChildMessenger is ChildMessengerInterface, Lockable, HasFinder {
+    uint32 public parentChainDomain;
+
+    event MessageSentToParent(
+        bytes data,
+        address indexed targetHub,
+        address indexed oracleSpoke,
+        uint32 parentChainDomain,
+        address parentMessenger
+    );
+    event MessageReceivedFromParent(
+        address indexed targetSpoke,
+        bytes dataToSendToTarget,
+        uint32 sourceDomain,
+        address sourceSender
+    );
+
+    /**
+     * @notice Only accept messages from a Nomad Replica contract
+     */
```

Comment:
might be better to use the `//` comment standard for this comment?

Reply:
done